docs/2.develop/relayers/relayer-gas-example.md

```diff
@@ -0,0 +1,152 @@
+---
+id: relayer-gas-example
+title: Multichain Relayer and Gas Station example
+sidebar_label: Relayer + Gas Station
+---
+import {CodeTabs, Language, Github} from "@site/src/components/codetabs"
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+In this article you'll learn how to run end-to-end tests on the entire Multichain Relayer system.  
+
+:::info Required tools
+
+For this tutorial, you'll need to have installed:
+
+ - [Multichain Relayer Server](https://github.com/near/multichain-relayer-server)
+ - [Gas Station Event indexer](https://github.com/near/gas-station-event-indexer)
+ - [NEAR CLI RS](https://github.com/near/near-cli-rs)
+
+:::
+
+## Setup
+
+Before you start testing, set up your local environment and install the Relayer server, the Event indexer and NEAR CLI.
+
+### Multichain Relayer server
+
+The main function of this server is interfacing with foreign chain RPCs sending both pre-signed funding transactions to cover gas and the actual pre-signed transaction once the funding is done.
+
+To run the [Multichain Relayer Server](https://github.com/near/multichain-relayer-server):
+
+1. Configure the Multichain Relayer by editing the [`config.toml`](https://github.com/near/multichain-relayer-server/blob/main/config.toml) file
+2. Start the multichain relayer server:
+   ```sh
+   cargo run
+   ```
+
+:::tip
+
+Find the Multichain Relayer server source code in [this GitHub repository](https://github.com/near/multichain-relayer-server).
+
+:::
+
+### Gas Station Event indexer
+
+The event indexer picks up events emitted from the [gas station contract](gas-station.md) used for generating signed foreign chain transactions and calls the multichain relayer `/send_funding_and_user_signed_txns` endpoint locally.
+
+To run the [Gas Station indexer](https://github.com/near/gas-station-event-indexer):
+
+1. Ensure you have the [Multichain Relayer Server](#multichain-relayer-server) running on `localhost:3030`
+2. Create and activate a Python virtual environment:
+   ```sh
+   pip install requirements.txt
+   ```
+
+3. Update the [`config.toml`](https://github.com/near/gas-station-event-indexer/blob/main/config.toml) configuration file with appropriate values
+   ```
+   network = "testnet"
+   # gas station contract account id
+   contract_id = "canhazgas.testnet"
+   ```
+
+4. Run the indexer:
+   ```sh
+   python3 gas-station-event-indexer.py
+   ```
+
+:::tip
+
+Find the Gas Station Event indexer source code in [this GitHub repository](https://github.com/near/gas-station-event-indexer).
+
+:::
+
+## Running tests
+
+1. Set the transaction details of the EVM transaction you want to send in [`generate_rlp_evm_txn.py`](https://github.com/near/multichain-relayer-server/blob/5b040611f2dc6c6b405b5ec00d5102e3cc27a65c/integration_tests/generate_rlp_evm_txn.py), run the script, and save the RLP hex string output.
```

Comment:
The gas station contract was updated to mainly support EIP-1559 transactions, so the note should just say `(Python and Rust output different hex RLP encoded transactions)` and the Rust and Python examples to generate transactions should use the [generate_eip1559_rlp_hex()](https://github.com/near/multichain-relayer-server/blob/feature/MVP_ethers/tests/tests.rs#L24) and [generate_rlp_encoded_transaction(is_eip_1559=true)](https://github.com/near/multichain-relayer-server/blob/feature/MVP_ethers/integration_tests/generate_rlp_evm_txn.py#L7) functions respectively.

Reply:
👍 note updated